modules/azure/linux-vm/extensions.tf

```diff
@@ -0,0 +1,39 @@
+# To get the type handler version run
+# az vm extension image list --location westeurope --output table
+resource "azurerm_virtual_machine_extension" "aad_ssh_login_for_linux" {
+  for_each = {
+    for vm in var.vm_config :
+    vm.vm_name => vm
+  }
+  name                 = "AADSSHLoginForLinux"
+  virtual_machine_id   = azurerm_linux_virtual_machine.vm[each.key].id
+  publisher            = "Microsoft.Azure.ActiveDirectory"
+  type                 = "AADSSHLoginForLinux"
+  type_handler_version = "1.0"
+
+}
+
+resource "azuread_group" "group_admins" {
+  display_name     = "sec-${var.rg_name}-vmadmin"
+  owners           = data.azuread_users.ad_group_owners.object_ids
+  security_enabled = true
+}
+
+resource "azuread_group" "group_users" {
+  display_name     = "sec-${var.rg_name}-vmuser"
+  owners           = data.azuread_users.ad_group_owners.object_ids
+  security_enabled = true
+}
+
+resource "azurerm_role_assignment" "role_admin" {
+  scope                = data.azurerm_resource_group.rg.id
```

Comment:
I'm not sure if we can do better, but in this group will give access to all VMs in this RG, not just the ones that is created by this module.

Reply:
Might be adjusted in future revisions